File: Offer/2273/main.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def kthLargest(self, root: TreeNode, k: int) -> int:
        res: int = 0
        def search(node: TreeNode):
            nonlocal res, k
            if node == None or k <= 0:
                return 
            search(node.right)
            k -= 1
            if k == 0:
                res = node.val
                return
            search(node.left)
            
        search(root)
        return res

File: Offer/2273/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func kthLargest(root *TreeNode, k int) (res int) {
	var search func(*TreeNode)
	search = func(node *TreeNode) {
		if node == nil || k <= 0 {
			return
		}
		search(node.Right)
		k--
		if k == 0 {
			res = node.Val
			return
		}
		search(node.Left)
	}
	search(root)
	return
}

// 给定一棵二叉搜索树，请找出其中第 k 大的节点的值。

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func kthLargest2(root *TreeNode, k int) (res int) {
	var search func(root *TreeNode)
	search = func(node *TreeNode) {
		if node == nil || k <= 0 {
			return
		}
		search(node.Right)
		k--
		if k == 0 {
			res = node.Val
			return
		}
		search(node.Left)
	}
	search(root)
	return
}
